### DISCUSSION TITLE
Mapping with fixed values

### DISCUSSION THREAD

**Discussion by fhalde** [apple/pkl#649] [General]
Is it possible to have fixed values in a Mapping? We have a consumer properties field which is a free-form map, however some values are not supposed to be amendable e.g. monitoring port

**Comment by bioball** [comment]
Are you in control of the configuration schema? If so, try separating them into different properties. For example:

```pkl
fixed port: Int = 5050

properties: Mapping<String, String|Int>
```

If, say, you aren't in control of the schema and need this to turn into a flat mapping, you can merge this before providing it to your target system.

For example, here, `port` and `extraProperties` as `hidden`, so that they don't show up in your output. And, you have `fixed properties` which is computed from the other two properties, which is non-hidden and shows up in the final output.

```pkl
hidden const port: Int = 5050

hidden extraProperties: Mapping<String, String|Int>

fixed properties: Mapping<String, String|Int> = (extraProperties) {
  ["port"] = port
}
```
